### PR TITLE
Fixes and updates for exposure strategies

### PR DESCRIPTION
**Pull request by MichalMaler**
Signed-off-by: Michal Maléř <mmaler@redhat.com>

* Adding ifevals for the Upstream-only content
* Fixing/Updating

### CODE REVIEW

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc [hunk 1]
```diff
@@ -22,9 +23,10 @@ To activate changes done to CheCluster YAML file, do one of the following:
 ----
 $ {prod-cli} server:deploy  --installer=operator --platform=... --che-operator-cr-patch-yaml=patch.yaml
 ----
-
++
+ifeval::["{project-context}" == "che"]
 patch.yaml should contain the following for `native` single-host exposure type:
```
Comment (rkratky) [inline modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc]:
```suggestion
* For the `native` single-host exposure type, use the following `patch.yaml` file:
```

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc [hunk 2]
```diff
@@ -22,9 +23,10 @@ To activate changes done to CheCluster YAML file, do one of the following:
 ----
 $ {prod-cli} server:deploy  --installer=operator --platform=... --che-operator-cr-patch-yaml=patch.yaml
```
Comment (rkratky) [inline modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc]: What's the expected value for `--platform=...`?
```suggestion
$ {prod-cli} server:deploy  --installer=operator --platform=... \
  --che-operator-cr-patch-yaml=patch.yaml
```

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc [hunk 3]
```diff
@@ -55,18 +57,21 @@ spec:
 ----
 
 
-
 * Update the YAML file properties of an already running cluster by executing the `{orch-cli} patch` command. For example:
 +
 [subs="+quotes,+attributes"]
 ----
 $ {orch-cli} patch checluster {prod-checluster} --type=json -p '[{"op": "replace", "path": "/spec/server/serverExposureStrategy", "value": "__single-host__"}] -n __<namespace-name>__'
```
Comment (rkratky) [inline modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc]:
```suggestion
$ {orch-cli} patch checluster {prod-checluster} --type=json \
  -p '[{"op": "replace", "path": "/spec/server/serverExposureStrategy", "value": "__single-host__"}] -n __<namespace-name>__'
```

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc [hunk 4]
```diff
@@ -55,18 +57,21 @@ spec:
...
+ or `default-host`
+endif::[]
+.
++
 If single-host type should be changed as well, command will look like:
-
++
 [subs="+quotes,+attributes"]
 ----
 $ {orch-cli} patch checluster {prod-checluster} --type=json -p '[{"op": "replace", "path": "/spec/server/serverExposureStrategy", "value": "single-host"},
```
Comment (rkratky) [inline modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc]:
```suggestion
$ {orch-cli} patch checluster {prod-checluster} --type=json \
  -p '[{"op": "replace", "path": "/spec/server/serverExposureStrategy", "value": "single-host"}, {"op": "replace", "path": "/spec/k8s/singleHostExposureType", "value": "gateway"}]' \
  -n _<namespace-name>_
```

Comment (rkratky) [reply modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc]: Remove the next line -- it was now appended to the command line.

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc [hunk 5]
```diff
@@ -14,6 +14,7 @@ link:https://docs.openshift.com/container-platform/latest/operators/olm-what-ope
 .Procedure
 
 When deploying {prod-short} using the Operator, configure the intended strategy by modifying the `spec.server.serverExposureStrategy` and  optionally `spec.k8s.singleHostExposureType` properties of the CheCluster Custom Resource object YAML file.
+
 To activate changes done to CheCluster YAML file, do one of the following:
 
 * Create a new cluster by executing the `{prod-cli}` command with applying a patch. For example:
```
Comment (rkratky) [inline modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc:20]:
```suggestion
* Create a new cluster by executing the `{prod-cli}` command with applying a patch. For example:
```

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc [hunk 6]
```diff
@@ -37,10 +39,10 @@ spec:
   k8s:
     singleHostExposureType: 'native'
 ----
-
-
+endif::[]
++
 For `gateway` type, patch.yaml should look as follows:
```
Comment (rkratky) [inline modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc]:
```suggestion
* For the `gateway` type, use the following `patch.yaml`:
```

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc [hunk 7]
```diff
@@ -55,18 +57,21 @@ spec:
 ----
 
 
-
 * Update the YAML file properties of an already running cluster by executing the `{orch-cli} patch` command. For example:
```
Comment (rkratky) [inline modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc:66]:
```suggestion
* Update the YAML file properties of an already running cluster by executing the `{orch-cli} patch` command. For example:
```

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc [hunk 8]
```diff
@@ -55,18 +57,21 @@ spec:
...
 +
 [subs="+quotes,+attributes"]
 ----
 $ {orch-cli} patch checluster {prod-checluster} --type=json -p '[{"op": "replace", "path": "/spec/server/serverExposureStrategy", "value": "__single-host__"}] -n __<namespace-name>__'
 ----
-
-Depending on the strategy used, replace the `_single-host_` option in the above example with `multi-host` or `default-host`.
-
++
+Depending on the strategy used, replace the `_single-host_` option in the above example with `multi-host`
```
Comment (rkratky) [inline modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc:74]: This sentence doesn't make sense when `"{project-context}" != "che"`.

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc [hunk 9]
```diff
@@ -55,18 +57,21 @@ spec:
...
+ifeval::["{project-context}" == "che"]
+ or `default-host`
+endif::[]
+.
++
 If single-host type should be changed as well, command will look like:
```
Comment (rkratky) [inline modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc]:
```suggestion
To change the `single-host` type, too, use:
```

Comment (rkratky) [reply modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc]: Is this possibility also valid when `"{project-context}" != "che"`?

**Review by rkratky — CHANGES_REQUESTED** [review]
Thanks, @MichalMaler. Some questions and suggestions added.

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc (hunk 2 above)
Comment (MichalMaler) [reply modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc]: Will be solved

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc (hunk 8 above)
Comment (MichalMaler) [reply modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc:74]: Hello Robert. Thanks for asking. The single-host mode is one of the new features for the CRW 2.5 release, thus it makes sense also for the downstream docs and ifeval is no more needed.

File: modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc (hunk 9 above)
Comment (MichalMaler) [reply modules/installation-guide/partials/proc_configuring-workspace-exposure-strategies-using-an-operator.adoc]: The single-host mode is one of the new features for the CRW 2.5 release, thus it makes sense also for the downstream docs and ifeval is no more needed.

**Review by MichalMaler — COMMENTED** [review]
I added Robert's feedback. LGTM now

**Review by rkratky — APPROVED** [review]

**Pull request merged**